### PR TITLE
fix: Unable to mock class with getters and setters

### PR DESCRIPTION
**Pull request by ebebbington**
https://stackoverflow.com/questions/71679061/mixin-class-in-typescript-cannot-be-mocked-by-rhum/71697623#71697623

### CODE REVIEW

File: src/mock_builder.ts [hunk 1]
```diff
@@ -52,8 +52,17 @@ export class MockBuilder<T> {
...
-      const desc = Object.getOwnPropertyDescriptor(original, property);
+      let desc = Object.getOwnPropertyDescriptor(original, property);
+      if (desc === undefined) {
+        // property is a getter or setter
+        desc = Object.getOwnPropertyDescriptor(
+          this.constructor_fn.prototype,
+          property,
+        );
+      }
+      console.log(`looking at ${property}, got desc`, desc);
```
Comment (Guergeiro) [inline src/mock_builder.ts:65]: Do we need this logs?

File: src/mock_builder.ts [hunk 2]
```diff
@@ -52,8 +52,17 @@ export class MockBuilder<T> {
     const original = new this.constructor_fn(...this.constructor_args);
 
     // Attach all of the original's properties to the mock
+    console.log("gonna try attach all props from original, to mock");
     this.getAllProperties(original).forEach((property: string) => {
-      const desc = Object.getOwnPropertyDescriptor(original, property);
+      let desc = Object.getOwnPropertyDescriptor(original, property);
+      if (desc === undefined) {
```
Comment (Guergeiro) [inline src/mock_builder.ts:58]: Is this different from `null`? If not, can't we simplify to `desc == null` to catch `null` as well?

**Review by Guergeiro — APPROVED** [review]
I approve this since it's basically single comments that do not affect the way the feature works.

File: src/mock_builder.ts (hunk 2 above)
Comment (ebebbington) [reply src/mock_builder.ts]: the vreturned value can be undefined, but i change the logic anyways, which inturn checks for null too

**Pull request merged**